src/Evolution/Systems/ScalarWave/Equations.hpp

```diff
@@ -68,4 +69,97 @@ struct ComputeDuDt {
       const tnsr::i<DataVector, Dim, Frame::Inertial>& d_pi,
       const tnsr::ij<DataVector, Dim, Frame::Inertial>& d_phi) noexcept;
 };
+
+/*!
+ * \brief Compute normal component of flux on a boundary.
+ *
+ * \f{align}
+ * F(\Psi) =& 0 \\
+ * F(\Pi) =& n^i \Phi_i \\
+ * F(\Phi_i) =& n_i \Pi
+ * \f}
+ */
+template <size_t Dim>
+struct ComputeNormalDotFluxes {
+  using return_tags =
+      typelist<Tags::NormalDotFlux<Pi>, Tags::NormalDotFlux<Phi<Dim>>,
+               Tags::NormalDotFlux<Psi>>;
+  using argument_tags = tmpl::list<Pi, Phi<Dim>>;
+  static void apply(gsl::not_null<Scalar<DataVector>*> pi_normal_dot_flux,
+                    gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*>
+                        phi_normal_dot_flux,
+                    gsl::not_null<Scalar<DataVector>*> psi_normal_dot_flux,
+                    const Scalar<DataVector>& pi,
+                    const tnsr::i<DataVector, Dim, Frame::Inertial>& phi,
+                    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+                        interface_unit_normal) noexcept;
+};
+
+/*!
+ * \brief Computes the upwind flux
+ *
+ * The upwind flux is given by:
+ * \f{align}
+ * F^*(\Psi) =& 0 \\
+ * F^*(\Pi) =& \frac{1}{2}\left(F(\Pi)_{\mathrm{int}} + F(\Pi)_{\mathrm{ext}}
+ *                    + \Pi_{\mathrm{int}} - \Pi_{\mathrm{ext}}\right) \\
+ * F^*(\Phi_i) =& \frac{1}{2} \left(F(\Phi_i)_{\mathrm{int}}
+ *                   + F(\Phi_i)_{\mathrm{ext}}
+ *                   + (n_i)_{\mathrm{int}} F(\Pi)_{\mathrm{int}}
+ *                   - (n_i)_{\mathrm{ext}} F(\Pi)_{\mathrm{ext}}\right)
+ * \f}
+ * where $F$ is the flux computed in ScalarWave::ComputeNormalDotFluxes
+ */
+template <size_t Dim>
+struct UpwindFlux {
+ private:
+  struct NormalTimesFluxPi {
+    using type = tnsr::i<DataVector, Dim, Frame::Inertial>;
+    static constexpr const char* const label = "UpwindFlux";
+  };
+
+ public:
+  using options = tmpl::list<>;
+  static constexpr OptionString help = {"Class help text"};
+
+  // clang-tidy: non-const reference
+  void pup(PUP::er& /*p*/) noexcept {}  // NOLINT
```

Comment:
Anything created from options needs to be pupable

Reply:
DomainCreators aren't pupable.  No idea if this class has to be, but creatability doesn't seem to imply it.

Reply:
True, this is stored in the global cache, though and aren't those created on Main then sent? (looking now)

Reply:
Yea, it looks like everything in the ConstGlobalCache is serializable: https://github.com/sxs-collaboration/spectre/blob/cc563ba35b8b1f55ae8bce5470e551ed525655f4/src/Parallel/Main.hpp#L197 or have I misunderstood that?

Reply:
Ah, yeah, global cache requiring serializability sounds right.

Reply:
Okay, cool! Since Will and I both weren't sure and are the people most familiar with that code, I'll add a comment explaining why the pup function is there.